### PR TITLE
Nguyenquyhy/16 min max toggle value usage

### LINKED ISSUE #16: Ability to put variables into Min/Max value ranges
I thought this setup would be pretty clever;

image

Putting `FUEL TANK LEFT MAIN QUANTITY` as the display value, and `FUEL TANK LEFT MAIN CAPACITY` as the max value, to auto-determine the range. Would be great if the min/max values could accept variables like this!

### PR DESCRIPTION
**Pull request by rmroc451**
closes #16 

depends on: https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/pull/76 (has changes in this PR since it isn't merged yet)

Added checkboxes to denote whether you want to use variable or constant in min/max.  I tried just parsing the value you put in, but it resolves the numeric value of the enumeration... I could do an upfront check if you are only using an int and not try to parse, but this seems like something we want to force them to make the decision on.

if you don't agree with that, I can go the route of not trying to parse the enum if the value they put in parses out to an in first. Let me know @nguyenquyhy .

demo profile:
[16_ MIN MAX VALUES FROM TOGGLE_VALUE.zip](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/5609292/16_.MIN.MAX.VALUES.FROM.TOGGLE_VALUE.zip)

### CODE REVIEW

**Comment by nguyenquyhy** [comment]
> if you don't agree with that, I can go the route of not trying to parse the enum if the value they put in parses out to an in first. Let me know @nguyenquyhy .

Yes I would prefer this (i.e. parse to number first), since we might extend it to something even more advanced later. Adding a checkbox now would complicate future changes. Also, I'm quite sure there is no SimConnect variable name that can match a number.

**Comment by rmroc451** [comment]
> > if you don't agree with that, I can go the route of not trying to parse the enum if the value they put in parses out to an in first. Let me know @nguyenquyhy .
> 
> Yes I would prefer this (i.e. parse to number first), since we might extend it to something even more advanced later. Adding a checkbox now would complicate future changes. Also, I'm quite sure there is no SimConnect variable name that can match a number.

Roger, Wilco.

**Comment by rmroc451** [comment]
@nguyenquyhy, were we waiting for something more on this to merge and close it out?

File: FlightStreamDeck.Logics/Actions/GenericGaugeAction.cs [hunk 1]
```diff
@@ -272,6 +293,14 @@ private async Task UpdateImage()
...
             }
         }
+
+        private string precision
+        {
+            get
+            {
+                return $"F{((settings.ValuePrecision?.Length ?? 0) > 0 ? settings.ValuePrecision : this.defaultSettings.ValuePrecision)}";
+            }
+        }
```
Comment (nguyenquyhy) [inline FlightStreamDeck.Logics/Actions/GenericGaugeAction.cs:303]: Don't need to change this, but C# has a nice syntax for get-only property that you might want to consider:

```
private string Precision 
    => $"F{((settings.ValuePrecision?.Length ?? 0) > 0 ? settings.ValuePrecision : this.defaultSettings.ValuePrecision)}";
```
This shows the intention quite nicely and doesn't waste too much space 😄.

**Review by nguyenquyhy — APPROVED** [review]

**Comment by nguyenquyhy** [comment]
> @nguyenquyhy, were we waiting for something more on this to merge and close it out?

Not waiting for anything, just that some random things come up last weekend, and I couldn't manage any time to take a look 😂.

**Comment by rmroc451** [comment]
I figured, didn't mean to rush you! Hope all is well!

**Pull request merged**